src/main/java/org/sonarsource/sonarlint/ls/AnalysisManager.java

```diff
@@ -491,22 +506,29 @@ private static Range position(ServerIssueLocation issue) {
   static String message(Issue issue) {
     if (issue.flows().isEmpty()) {
       return issue.getMessage();
+    } else if (issue.flows().size() == 1) {
+      return buildMessageWithPluralizedSuffix(issue.getMessage(), issue.flows().get(0).locations().size(), ITEM_LOCATION);
     } else if (issue.flows().stream().allMatch(f -> f.locations().size() == 1)) {
       int nbLocations = issue.flows().size();
-      return String.format(MESSAGE_WITH_PLURALIZED_SUFFIX, issue.getMessage(), nbLocations, pluralize(nbLocations, "location"));
+      return buildMessageWithPluralizedSuffix(issue.getMessage(), nbLocations, ITEM_LOCATION);
     } else {
       int nbFlows = issue.flows().size();
-      return String.format(MESSAGE_WITH_PLURALIZED_SUFFIX, issue.getMessage(), nbFlows, pluralize(nbFlows, "flow"));
+      return buildMessageWithPluralizedSuffix(issue.getMessage(), nbFlows, ITEM_FLOW);
     }
   }
 
   static String message(ServerIssue issue) {
-    int nbFlows = issue.getFlows().size();
-    return String.format(MESSAGE_WITH_PLURALIZED_SUFFIX, issue.getMessage(), nbFlows, pluralize(nbFlows, "flow"));
+    if (issue.getFlows().isEmpty()) {
+      return issue.getMessage();
+    } else if (issue.getFlows().size() == 1) {
+      return buildMessageWithPluralizedSuffix(issue.getMessage(), issue.getFlows().get(0).locations().size(), ITEM_LOCATION);
+    } else {
+      return buildMessageWithPluralizedSuffix(issue.getMessage(), issue.getFlows().size(), ITEM_FLOW);
+    }
```

Comment:
Here we don't consider the case where we can several flows having each a single location (as we do for live issues), but maybe it's not relevant for taint vulnerabilities

Reply:
Indeed, this case should only happen with local issues.